Client/Frontend/Login Management/BreachAlertsManager.swift

```diff
@@ -30,50 +30,57 @@ final public class BreachAlertsManager {
     static let detailColor = UIColor(red: 0.59, green: 0.11, blue: 0.11, alpha: 1.00)
     static let monitorAboutUrl = URL(string: "https://monitor.firefox.com/about")
     var breaches = Set<BreachRecord>()
-    var breachAlertsClient: BreachAlertsClientProtocol
-
-    init(_ client: BreachAlertsClientProtocol = BreachAlertsClient()) {
-        self.breachAlertsClient = client
+    var client: BreachAlertsClientProtocol
+    var profile: Profile!
+    private lazy var cacheURL: URL = {
+        return URL(fileURLWithPath: (try? self.profile.files.getAndEnsureDirectory())!, isDirectory: true).appendingPathComponent("breaches.json")
+    }()
+    init(_ client: BreachAlertsClientProtocol = BreachAlertsClient(), profile: Profile) {
+        self.client = client
+        self.profile = profile
     }
 
     /// Loads breaches from Monitor endpoint using BreachAlertsClient.
     ///    - Parameters:
     ///         - completion: a completion handler for the processed breaches
     func loadBreaches(completion: @escaping (Maybe<Set<BreachRecord>>) -> Void) {
-        self.breachAlertsClient.fetchData(endpoint: .breachedAccounts) { maybeData in
-            guard let data = maybeData.successValue else {
-                completion(Maybe(failure: BreachAlertsError(description: "failed to load breaches data")))
+        if FileManager.default.fileExists(atPath: self.cacheURL.path) {
+            guard let fileData = FileManager.default.contents(atPath: self.cacheURL.path) else {
+                completion(Maybe(failure: BreachAlertsError(description: "failed to get data from breach.json")))
                 return
             }
-            guard let decoded = try? JSONDecoder().decode(Set<BreachRecord>.self, from: data) else {
-                completion(Maybe(failure: BreachAlertsError(description: "JSON data decode failure")))
+
+            // 1. check the last time breach endpoint was accessed
+            guard let dateLastAccessedString = profile.prefs.stringForKey(BreachAlertsClient.dateKey) else {
                 return
             }
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateFormat = "E, d MMM yyyy HH:mm:ss zzz"
+            guard let dateLastAccessed = dateFormatter.date(from: dateLastAccessedString) else { return }
+
+            let timeUntilNextUpdate = 60.0 * 60.0 * 24.0 * 3.0 // 3 days in seconds
+            let shouldUpdateDate = Date(timeInterval: timeUntilNextUpdate, since: dateLastAccessed)
+
+            // 2a. if 3 days have passed since last update...
+            if shouldUpdateDate.timeIntervalSinceNow >= timeUntilNextUpdate {
```

Comment:
`guard` instead pls

Reply:
Same issue as the other `if` above, there is an `else` attached

Reply:
Here is the flatter version I was thinking of:
```
func loadBreaches(completion: @escaping (Maybe<Set<BreachRecord>>) -> Void) {
        guard FileManager.default.fileExists(atPath: self.cacheURL.path) else {
            // first time loading breaches, so fetch as normal and hand off
            self.fetchAndSaveBreaches(completion)
        }

        guard let fileData = FileManager.default.contents(atPath: self.cacheURL.path) else {
            completion(Maybe(failure: BreachAlertsError(description: "failed to get data from breach.json")))
            Sentry.shared.send(message: "BreachAlerts: failed to get data from breach.json")
            return
        }

        // 1. check the last time breach endpoint was accessed
        guard let dateLastAccessed = profile.prefs.timestampForKey(BreachAlertsClient.etagDateKey) else {
            return
        }

        let timeUntilNextUpdate = UInt64(60 * 60 * 24 * 3) // 3 days in seconds
        let shouldUpdateDate = dateLastAccessed + timeUntilNextUpdate

        // 2a. if 3 days have passed since last update...
        guard Date.now() < shouldUpdateDate else {
            // 2b. else, no need to refetch. decode local data and hand off
            decodeData(data: fileData, completion)
        }

        // 3. check if the etag is different
        self.client.fetchEtag(endpoint: .breachedAccounts, profile: self.profile) { etag in
            guard let etag =  etag else { return }
            let savedEtag = self.profile.prefs.stringForKey(BreachAlertsClient.etagKey)

            // 4. if it is, refetch the data and hand entire Set of BreachRecords off
            if etag != savedEtag {
                self.fetchAndSaveBreaches(completion)
            }
        }
    }
```